pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -581,3 +565,77 @@ func installNetworkPolicies(instance *servingv1alpha1.KnativeServing) error {
 	}
 	return nil
 }
+
+func exposeMetricToServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	// getOperatorNamespace return namespace where knative-serving-operator has been installed
+	namespace, err := getOperatorNamespace()
+	if err != nil {
+		log.Info("no namespace defined, skipping ServiceMonitor installation for the operator")
+		return nil
+	}
+	log.Info("exposing metric for installed knative version")
+	knativeVersion.WithLabelValues(version.Version, namespace).Set(float64(time.Now().Unix()))
+
+	log.Info("Installing Service Monitor for Operator")
+	const path = "deploy/resources/monitoring/operator_service_monitor.yaml"
+	if err := createServiceMonitor(instance, namespace, path); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getOperatorNamespace() (string, error) {
+	ns, found := os.LookupEnv(knativeServingInstalledNamespace)
+	if !found {
+		return "", fmt.Errorf("%s must be set", knativeServingInstalledNamespace)
+	}
+	return ns, nil
+}
+
+func addMonitoringLabelToNamespace(namespace string) error {
+	ns := &v1.Namespace{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Name: namespace}, ns); err != nil {
+		return err
+	}
+	if ns.Labels == nil {
+		ns.Labels = map[string]string{}
+	}
+	ns.Labels[monitoringLabel] = "true"
+	if err := api.Update(context.TODO(), ns); err != nil {
+		log.Error(err, fmt.Sprintf("could not add label %q to namespace %q", monitoringLabel, namespace))
+		return err
+	}
+	return nil
+}
+
+func createServiceMonitor(instance *servingv1alpha1.KnativeServing, namespace, path string) error {
+	if serviceMonitorExists, err := serviceMonitorExists(namespace); err != nil {
+		return err
+	} else if !serviceMonitorExists {
+		log.Info("ServiceMonitor CRD is not installed. Skip to install ServiceMonitor")
+		return nil
+	}
+	// Add label openshift.io/cluster-monitoring to namespace
+	if err := addMonitoringLabelToNamespace(namespace); err != nil {
+		return err
+	}
+	// Install ServiceMonitor
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		log.Error(err, "Unable to create ServiceMonitor install manifest")
+		return err
+	}
+	transforms := []mf.Transformer{mf.InjectOwner(instance)}
```

Comment:
I am wondering that we would clean up the existing Role&RoleBinding whenever we uninstall the serving-operator. I mean that the same problem with [this](https://github.com/openshift-knative/knative-serving-operator/pull/49) happened on Role&Rolebiding. If so, I think we should change the Role&Role name (=prometheus-k8s).